test-network-k8s/docs/CA.md

```diff
@@ -0,0 +1,281 @@
+  and identity MSPs.
+  
+
+- Certificate organization and [Folder Structure](https://hyperledger-fabric-ca.readthedocs.io/en/latest/deployguide/use_CA.html#folder-structure-for-your-org-and-node-admin-identities)
+  strictly adheres to the best practices and guidelines recommended by the CA Deployment Guide. 
+
+
+- The `cryptogen` anti-pattern is **strictly forbidden**.  All TLS and MSP enrollments are constructed using the CA 
+  registration and enrollment REST services, coordinated by calls to `fabric-ca-client` running directly on the 
+  CA pods.  When working with certificates, the fabric CA client ONLY has visibility to the organization's local volume 
+  storage. 
+
+
+- TLS CA configuration and certificates are maintained in each org's persistent volume at `/var/hyperledger/fabric-tls-ca-server`
+
+
+- ECert CA configuration and certificates are maintained in each org's persistent volume at `/var/hyperledger/fabric-ca-server`
+
+
+- fabric-ca-client configuration and certificates are maintained in each org's persistent volume at `/var/hyperledger/fabric-ca-client`
+
+
+- ECert and MSP data structures are maintained in each org's persistent volume at `/var/hyperledger/fabric/organizations`
+
+
+
+### Future Enhancements: 
+
+- **_Bring your own Certificates_** :  It would be nice to boostrap the network using a single, top-level signing authority, 
+  rather than generating self-signed certificates when the system is bootstrapped.  Ideally this will be realized by 
+  introducing an [Intermediate CA](https://hyperledger-fabric-ca.readthedocs.io/en/latest/deployguide/ca-deploy-topology.html#when-would-i-want-an-intermediate-ca)
+  and/or alternate signing chains backed by formal (e.g. letsencrypt, Thawte, Verisign, etc.) certificate authorities. 
+
+
+- **_Dual Headed CAs_** : In practice, juggling two distinct deployments between TLS and ECert servers adds little 
+  functional value.  It would be nice to simplify the configuration, deployment, and bootstrapping scripts such that 
+  each org manages a single, dual-headed CA capable of responding to both TLS as well as ECert enrollmnent rerquests.
+  
+
+- **_Time-Bomb Certificates_** : By default the certificates issued by the test network are valid for 1 (one) year.  For 
+  lightweight or adhoc testing, this is fine.  But when applied to production deployments, certificate expiry is a 
+  real operational challenge.  For instance, it is possible to soft-lock a Fabric network when all system certificates
+  expire _en-masse_ - it's impossible to re-establish a consensus and renew the certificates!
+  
+
+- **_Mutual TLS_** : Server-side TLS is a minimum, but the addition of client-side TLS certificates will help fully 
+  secure all TCP channels within the Fabric network.  
+
+
+- **_Bugs_** : `./network up` currently goes through the process of bootstrapping a fabric network from scratch, but 
```

Comment:
ah - probably worth copying this up to a higher level,.

Reply:
Agree.  This CAN NOT be even a "production-like" network if running `network up` twice in succession will soft-lock the network.    This either needs to be fixed (preferably), or right up at the top of page 1 in glaring, double-<BLINK> text tags.